File: Loenn/entities/itemArea.lua
local xnaColors = require("consts.xna_colors")
local green = xnaColors.Green

local itemArea = {}

itemArea.name = "PuzzleIslandHelper/EscapeEntities/ItemArea"
itemArea.fillColor = {green[1] * 0.3, green[2] * 0.3, green[3] * 0.3, 0.6}
itemArea.borderColor = {green[1] * 0.8, green[2] * 0.8, green[3] * 0.8, 0.8}
itemArea.placements = {
    name = "Item Area",
    data = {
        width = 8,
        height = 8,
    }
}
itemArea.depth = 0

return itemArea

File: Entities/EscapeRoomEntities/ItemArea.cs
using Celeste.Mod.Entities;
using Celeste.Mod.PuzzleIslandHelper.Entities.Programs;
using FrostHelper;
using Microsoft.Xna.Framework;
using Microsoft.Xna.Framework.Input;
using Monocle;
using System;
using System.CodeDom;
using System.Collections;
using System.Collections.Generic;
using System.Linq;

namespace Celeste.Mod.PuzzleIslandHelper.Entities.EscapeRoomEntities
{

    [CustomEntity("PuzzleIslandHelper/EscapeEntities/ItemArea")]
    [Tracked]
    public class ItemArea : Entity
    {
        public string ItemName;
        public bool Disabled;
        public override void Render()
        {
            base.Render();
        }
        public ItemArea(EntityData data, Vector2 offset) : base(data.Position + offset)
        {
            Collider = new Hitbox(data.Width, data.Height);
        }
        public override void Awake(Scene scene)
        {
            base.Awake(scene);
        }
        public override void Update()
        {
            base.Update();
            if (Scene is not Level level || Disabled)
            {
                return;
            }
            if (CollideCheck<Player>())
            {
                SetItemUsability(level);
            }
        }
        public void SetItemUsability(Level level)
        {
            foreach (Item item in level.Tracker.GetComponents<Item>())
            {
                item.CanUse = item.Name == ItemName;
            }
        }
    }
}